target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-rtl30vw.dts

```diff
@@ -0,0 +1,407 @@
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+	cs-gpios = <&tlmm 54 0>, <&tlmm 59 0>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor", "n25q128a11";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		spi-max-frequency = <24000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:SBL1";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+			partition@40000 {
+				label = "0:MIBIB";
+				reg = <0x40000 0x20000>;
+				read-only;
+			};
+			partition@60000 {
+				label = "0:QSEE";
+				reg = <0x60000 0x60000>;
+				read-only;
+				};
+			partition@c0000 {
+				label = "0:CDT";
+				reg = <0xc0000 0x10000>;
+				read-only;
+			};
+			partition@d0000 {
+				label = "0:DDRPARAMS";
+				reg = <0xd0000 0x10000>;
+				read-only;
+			};
+			partition@e0000 {
+				label = "0:APPSBLENV";
+				reg = <0xe0000 0x10000>;
+				read-only;
+			};
+			partition@f0000 {
+				label = "0:APPSBL";
+				reg = <0xf0000 0x80000>;
+				read-only;
+			};
+			partition@170000 {
+				label = "0:ART";
+				reg = <0x170000 0x10000>;
+				read-only;
+			};
+			partition@180000 {
+				label = "0:BOOTCONFIG";
+				reg = <0x180000 0x10000>;
+				read-only;
+			};
+		};
+	};
+
+	flash@1 {
+		compatible = "spi-nand","spinand,mt29f";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+		spi-max-frequency = <24000000>;
+
+		/*
+		 * Factory U-boot looks in 0:BOOTCONFIG partition for active
+		 * partitions settings and mangle partition config. So kernel
+		 * /kernel_1 and rootfs/rootfs_1 pairs can be swaped.
+		 * It isn't a problem but we never can be sure where OFW put
+		 * factory images.
+		 */
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "kernel";
+				reg = <0x0 0x400000>;
+			};
+			partition@400000 {
+				label = "rootfs";
+				reg = <0x400000 0x2000000>;
+			};
+			partition@2400000 {
+				label = "kernel_1";
+				reg = <0x2400000 0x400000>;
+			};
+			partition@2800000 {
+				label = "rootfs_1";
+				reg = <0x2800000 0x2000000>;
+			};
+			partition@4800000 {
+				label = "ubifs";
+				reg = <0x4800000 0x3800000>;
+			};
+
+		};
+	};
+};
+
+&blsp1_uart1 {
+	pinctrl-0 = <&serial_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+};
+
+&cryptobam {
+	status = "ok";
+};
+
+&tlmm {
+	serial_pins: serial_pinmux {
+		mux {
+			pins = "gpio60", "gpio61";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+
+	spi_0_pins: spi_0_pinmux {
+		pinmux {
+			function = "blsp_spi0";
+			pins = "gpio55", "gpio56", "gpio57";
+		};
+		pinmux_cs {
+			function = "gpio";
+			pins = "gpio54", "gpio59";
+		};
+		pinconf {
+			pins = "gpio55", "gpio56", "gpio57";
+			drive-strength = <12>;
+			bias-disable;
+		};
+		pinconf_cs {
+			pins = "gpio54", "gpio59";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+};
+
+&wifi0 {
```

Comment:
No custom BDF-s for radios?

Reply:
```
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_YA131_wifi1.bin
/lib/firmware/IPQ4019/hw.1/athwlan.codeswap.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK04_2G.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK03_wifi1.bin
/lib/firmware/IPQ4019/hw.1/utf.codeswap.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK01_2G.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK04_5G.bin
/lib/firmware/IPQ4019/hw.1/athwlan.bin
/lib/firmware/IPQ4019/hw.1/boarddata_0.bin
/lib/firmware/IPQ4019/hw.1/utf.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK04_2G_neg_pwr.bin
/lib/firmware/IPQ4019/hw.1/boarddata_1.bin
/lib/firmware/IPQ4019/hw.1/otp.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK01_5G.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_Y9803_wifi0.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_YA131_wifi0.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK04_5G_neg_pwr.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_DK03_wifi0.bin
/lib/firmware/IPQ4019/hw.1/boardData_1_0_IPQ4019_Y9803_wifi1.bin
```

Could You help? Does any looks like custom file?

Reply:
It will be one of those for 2GHz and one for 5GHz.
Easiest way to see which one is used to look into the bootlog, you will be able to see which file is loaded for which radio.

Reply:
Thank You. 

````
[   27.941604] ol_transfer_bin_file: Board Data File download to address=0xc0000 file name=IPQ4019/hw.1/boardData_1_0_IPQ4019_Y9803_wifi0.bin
[   28.161380] ath9k a000000.wifi: Firmware loaded from user helper succesfully
[   31.148593] ol_transfer_bin_file: Board Data File download to address=0xc0000 file name=IPQ4019/hw.1/boardData_1_0_IPQ4019_Y9803_wifi1.bin
[   31.161248] ath9k a800000.wifi: Firmware loaded from user helper succesfully
````

Got it. I will add it to PR.

Reply:
From the ML post:
""bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=AVM-FRITZBox-4040", so
I didn't send it."

Please sent it too, even if it's a duplicate. Ideally, the board-2.bin format can be extended to handle duplicates, but qualcomm will do that on its own once they notice that there's a demand.

Reply:
@chunkeey 
 Done. I sent v2 with both files.